### ISSUE TITLE
Replacing current strategies with a smaller subset

### ISSUE THREAD

**Issue by htwangtw** [SIMEXP/load_confounds#143]
As part of the discussion at Nilern-Nibable dev days 2021.
@edickie suggested we need documentation on the straw man strategies in Ciric 2017, to warn users not to use them. Or we can consider deleting them entirely?

**Comment by pbellec** [comment]
I would personally be in favour of minimizing dramatically the number of strategies, and offering a short rationale of why (and when) we think those strategies can be helpful.

**Comment by htwangtw** [comment]
I am thinking removing Params2 and perhaps models systematically performing worse than it in the Ciric paper? 
That would make the list:
- Params2 
- Params6 
- Params24 (low performance and losing high number of DoF)
- TempCompCor 

I am now preparing a big PR to nilearn with the code from current master branch. Do we want strategies removed when migrate to nilearn or do it here as well?
Let me know what do you think.

**Comment by pbellec** [comment]
I suggest to have only a few strategies: `Minimal`, `Scrubbing`, `CompCor` and `ICAAROMA`, and expose a limited number of parameters for those strategies (like `global` which one may want to enable in any strategy). Also we need a strategy for the nilearn PR to acknowledge all contributors. More details below.

The names selected in Ciric 2017 are very hard to follow, and somewhat misleading. For example `Params9` has six motion + wm + csf + global, but because it includes the slow time drifts, it features more than 9 regressors. Also, we have the ability to control the scope of parameters exposed to users for a given strategy, but we don't need to make that scope as narrow as it is now. Proposal:
 * `Minimal` that's for data with minimal motion. Only includes motion parameters, wm and csf, with the option to add global. All parameters related to these confounds are exposed (like degree of expansion). 
 * `Scrubbing`  like minimal, but with scrubbing. Pros: Actual impact on data is pretty limited, but good and offers the most control on what's being discarded. Cons: high loss of degrees of freedom. 
 * `CompCor`  we would expose all compcor-related parameters (include `anat` or `temp`, with `anat` as default), as well as the ability to expand the motion model (by default use expansion). This could be used to implement either `TempCompcor` or `AnatCompcor`. Pros: large impact of denoising, efficient denoising, controlled loss of degrees of freedom. Cons: low control on what is being discarded (who knows what signal actually show up in the PCA for a given subject). 
 * `ICAAROMA`: this one would be the non-aggressive version with ability to add `global` (default no `global`). The aggressive variant is a bit of an odd :duck: and can be implemented easily with the `Confounds` if a user really wants it. Pros: pretty similar to CompCor, with better control of discarded components (those can be visually reviewed even though this is time consuming. Cons: may require retraining the noise detector and also requires to believe that ICA does efficiently separate noise from signal, which is not that clear, and the quality of separation may also vary substantially across subjects. 

Of course we would make the doc a bit less concise and more evidence-based (with references), but that's the gist of it I think. Regarding the PR, I would have really liked being able to keep the history of commits from the load_confounds repo. We could reach out to the nilearn team - they've done something similar recently with the merge of their GLM tools which lived in a separate repo. An alternative would be to have a multi-authored pull request https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors

**Comment by htwangtw** [comment]
These suggestions are awesome! I forgot about the comment on readability. Thank you for bringing that point up. I have a question related to the implementations of spike regressor (Satterthwaite et al 2013; model 7 in Ciric rt al. 2017) and scrubbing shares a lot of procedural similarity. Scrubbing used `framewise_displacement` to evaluate movement and spike regressor uses `rmsd`. The only difference is the final step, where scrubbing discards the detected volumes, and despike add those spike regressors as part of the confounds matrix. Would it make sense to have these two strategies implemented as one, with some options exposed to users. (I am thinking about something like this: metric: {`fd`, `rmsd`}, censor: {`temporal_mask`, `regressor`}.)

I looked into the merge history of nistats: looked like they have the whole repo in `nilearn` first to retain the history before the full integration.
This is possibly the first [commit](https://github.com/nilearn/nilearn/tree/1fd929d6b1efb65b0b44160f9e29465e6a536463) of the merge.  
Relevant thread on [stack overflow](https://stackoverflow.com/questions/1683531/how-to-import-existing-git-repository-into-another/8396318#8396318)
I will comment on the relevant nilearn issue and see what they think.

**Comment by pbellec** [comment]
I like your suggestion of adding a parameter to `Scrubbing` that would select `fd` or `rmsd` or `fd+dvars` (default). 

Regarding the `temporal_mask` vs `regressor`, as this changes the format of the output, I'd personally be leaning more having two separate methods `load` and `load_with_mask`. This would signal that the API is fundamentally different. But I don't have a strong opinion on this.

**Comment by effigies** [comment]
I'm not entirely clear on what's being proposed. Are `Minimal` and friends parameterizable classes that serialize to a set of confounds (or functions that return a list of confound names)?

I agree in principle with not promoting Ciric et al strategies to their own aliases. I do think it would be good for an example to show how to reconstruct any of the Ciric strategies, demonstrating the flexibility of the approach.

I suspect this isn't the response you were looking for... feel free to ask more pointed questions.

**Comment by effigies** [comment]
Got it. Well this all seems reasonable, then. I can't readily think of others that fMRIPrep provides.
